bfsbdump

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python3
+
+"""
+Dump BlueField secure boot status information.
+"""
+
+import argparse
+import struct
+import binascii
+import os
+import sys
+import subprocess
+
+LifeCycleState = [ 'Production', 'Secure', 'Non-Secure', 'RMA' ]
+
+def GetPlatformName():
+    Out = subprocess.run(['bfhcafw', 'mcra', '0xf0014.0:16'], stdout=subprocess.PIPE)
+    DeviceId = Out.stdout[:-1].decode('utf-8')
+    if DeviceId == '0x00000211':
+        return 'BlueField1'
+    elif DeviceId == '0x00000214':
+        return 'BlueField2'
+    elif DeviceId == '0x0000021c':
+        return 'BlueField3'
+    else:
+        printf('Error! can not read device id')
+        sys.exit (1)
+
+def CountSetBits(n):
+    if n == 0:
+        return 0
+    else:
+        return (n & 1) + CountSetBits(n >> 1)
```

Comment:
This can be a one-liner:
```python
def CountSetBits(n):
    return bin(n).count("1")
```

It's also faster, not that it matters much for the small amount of data it's used for:
```pycon
>>> import timeit
>>> setup="""
... def CountSetBits(n):
...     if n == 0:
...             return 0
...     else:
...             return (n&1)+CountSetBits(n>>1)
... """
>>> timeit.timeit(stmt="CountSetBits(123456)", setup=setup, number=1000000)
4.13406078889966

>>> setup="""
... def CountSetBits(n):
...     return bin(n).count("1")
... """
>>> timeit.timeit(stmt="CountSetBits(123456)", setup=setup, number=1000000)
0.43963913060724735
```

Reply:
Good to know that. Thanks!